processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -98,18 +98,20 @@
     }
 
     @Override
+    @Nullable
     public String fromByteBuffer(final ByteBuffer buffer, final int numBytes)
     {
-      return StringUtils.fromUtf8(buffer, numBytes);
+      if (numBytes < 0) {
+        // nulBytes will be -1 for null values.
```

Comment:
Can you add this info to the format specification at the top of GenericIndexed as well?

Maybe use a constant for the -1 here with a name that indicates it's the marker for a null value

Reply:
done.